cmd/cleanup-jobs.go

```diff
@@ -0,0 +1,62 @@
+package cmd
+
+import (
+	"context"
+	"log"
+	"sync"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// CleanupJobs removes all failed jobs in a given namespace
+var CleanupJobs = &cobra.Command{
+	Use:   "cleanupjobs",
+	Short: "remove all failed jobs in a given namespace",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		namespace, _ := cmd.Flags().GetString("namespace")
+
+		p := getPlatform(cmd)
+		clientSet, err := p.GetClientset()
+
+		check(err)
+
+		// gather the list of jobs from a namespace
+		jobs, err := clientSet.BatchV1().Jobs(namespace).List(context.TODO(), metav1.ListOptions{})
+
+		// WaitGroup to synchronize go routines execution
+		wg := sync.WaitGroup{}
+
+		// loop through the list of jobs found
+		for _, j := range jobs.Items {
+			wg.Add(1)
+
+			go func(j v1.Job, clientSet *kubernetes.Clientset) {
+
+				// if job status is not equal to "succeeded", delete the job
+				if j.Status.Failed == 0 {
```

Comment:
This check seems wrong - if the job failed, then the count of failed pods would be > 0?  I would also recommend checking though JobConditions or similar that the job has completed before attempting to delete

Reply:
You're right! 👍
I was doing some tests and forgot to change it back.